### PR TITLE
Housekeeping

### PR DESCRIPTION
**Pull request by julienmartinlevrai**
* Upgrade solc to ~0.6.12
* Upgrade dss, dss-gem-joins and dss-interfaces

**Note:** testing should be done on Kovan

### CODE REVIEW

**Review by brianmcmichael — APPROVED** [review]

**Review by brianmcmichael — APPROVED** [review]

**Review by iamchrissmith — APPROVED** [review]
lgtm tests pass locally against kovan

File: src/RwaConduit.sol [hunk 1]
```diff
@@ -13,7 +18,7 @@
 // You should have received a copy of the GNU Affero General Public License
 // along with this program.  If not, see <https://www.gnu.org/licenses/>.
 
-pragma solidity 0.5.12;
+pragma solidity ~0.6.12;
```
Comment (naszam) [inline src/RwaConduit.sol]: note: introducing the `~` modifier should be equivalent 0.6.12 <= n < 0.7.0.
as discussed in chat, I don't see any benefit in using it, sharing here as well for general discussion.

**Comment by naszam** [comment]
@julienmartinlevrai I've just noticed that the old license file (`COPYING`) is present alongside with the new one.
Following https://github.com/makerdao/dss/pull/254, should we remove the old one as part of the Housekeeping as well here?
cc: @brianmcmichael

File: src/RwaConduit.sol (hunk 1 above)
Comment (naszam) [reply src/RwaConduit.sol]: I would remove it if not necessary

**Comment by naszam** [comment]
checking `dss-gem-joins` upgrade, It seems not synced with latest on master, consider `dapp upgrade dss-gem-joins` again to reflect those changes.
same for `dss-interfaces`.

**Review by naszam — APPROVED** [review]
Checked:

- Pragma Changes => lock to `0.6.12`
- `dapp upgrade` deps
	- `lib/dss` => `c666ab1` (latest on master)
	- `lib/dss-gem-joins` => `452cb62` (latest on master)
	- `lib/dss-interfaces` => `a82bc3a` (latest on master)
	- other deps => latest on master
- License Changes Look Good
- Test PASS (Kovan)

LGTM

**Pull request merged**